### PR TITLE
encode Segment.Map attribute

### PR DESCRIPTION
**Pull request by andyborn**
Hi @grafov have added encoding for Segment.Map to address https://github.com/grafov/m3u8/issues/93

### CODE REVIEW

**Comment by bradleyfalzon** [comment]
Thanks, this looks good to me, would you be able to provide a unit test for it as well?

Also, it appears we're reading `EXT-X-MAP` on both playlist and segments, is that correct? There was only a problem in writing the tag per segment?

File: writer_test.go [hunk 1]
```diff
@@ -313,6 +313,7 @@ func TestSetMapForMediaPlaylist(t *testing.T) {
 	if e != nil {
 		t.Errorf("Set map to a media playlist failed: %s", e)
 	}
+	// fmt.Println(p.Encode().String())
```
Comment (bradleyfalzon) [inline writer_test.go]: If you're able to write the relevant tests, can you remove this comment?

File: writer_test.go [hunk 2]
```diff
@@ -566,7 +567,7 @@ func TestNewMasterPlaylistWithAlternatives(t *testing.T) {
 	if m.ver != 4 {
 		t.Fatalf("Expected version 4, actual, %d", m.ver)
 	}
-	fmt.Printf("%v\n", m)
+	// fmt.Printf("%v\n", m)
```
Comment (bradleyfalzon) [inline writer_test.go]: If you're able to write the relevant tests, can you remove this comment?

File: writer_test.go [hunk 3]
```diff
@@ -298,8 +298,23 @@ func TestSetDefaultKeyForMediaPlaylist(t *testing.T) {
...
 // Create new media playlist
+// Set default map
+func TestSetDefaultMapForMediaPlaylist(t *testing.T) {
+	p, e := NewMediaPlaylist(3, 5)
+	if e != nil {
+		t.Fatalf("Create media playlist failed: %s", e)
+	}
+	p.SetDefaultMap("https://example.com", 1000*1024, 1024*1024)
+
+	expected := "EXT-X-MAP:URI=\"https://example.com\",BYTERANGE=1024000@1048576"
```
Comment (bradleyfalzon) [inline writer_test.go]: You can rewrite with backticks so you don't need to escape the quotes, eg

```
expected := `#EXT-X-MAP:URI="https://example.com",BYTERANGE=1024000@1048576`
```

File: writer_test.go [hunk 4]
```diff
@@ -313,6 +328,43 @@ func TestSetMapForMediaPlaylist(t *testing.T) {
 	if e != nil {
 		t.Errorf("Set map to a media playlist failed: %s", e)
 	}
+
+	expected := "EXT-X-MAP:URI=\"https://example.com\",BYTERANGE=1024000@1048576\n#EXTINF:5.000,\ntest01.ts"
```
Comment (bradleyfalzon) [inline writer_test.go]: Alternatively:

```
expected := `EXT-X-MAP:URI="https://example.com",BYTERANGE=1024000@1048576
#EXTINF:5.000,
test01.ts`

Comment (andyborn) [inline writer_test.go]: thanks, have updated test expectations to string literals where possible.  hopefully good to merge now.

**Comment by bradleyfalzon** [comment]
Thanks for the contribution Andy and the additional tests. Much appreciated.

**Pull request merged**